### ISSUE TITLE
Due to how NextJs works with NextResponse.next, this does not work at all

### ISSUE THREAD

**Issue by cole21771** [labd/nextjs-basic-auth-middleware#27]
If you press cancel on the basic auth prompt it loads the page anyways and if you press it enough times the browser stops prompting you and then you are free to roam the "restricted" page. RIP

**Comment by happylolonly** [comment]
@blurrah yes, I also tried, it is a problem, can you fix it?

**Comment by rbasto1** [comment]
This worked for us:

`middleware.ts` on the same level as `pages/`:
```
import { createNextAuthMiddleware } from 'nextjs-basic-auth-middleware';

export const middleware = createNextAuthMiddleware({
  users: [{ name: 'user', password: 'super-secret' }],
});

export const config = {
  matcher: ['/(.*)', '/_next/static/chunks/(.*)'],
};
```

**Comment by blurrah** [comment]
Yes you are completely right and this is problematic, means we'll need to use the rewrite to an API route approach.

I'll start working on that approach right now and will deprecate the built in path matcher for the official middleware path matcher.

**Comment by blurrah** [comment]
This is now available in 2.0.0. Do note that API's have been changed and you'll need to update your code :)

**Comment by cole21771** [comment]
Holy crap that worked! 🙌🙌🙌🙌 Thank you so much @blurrah!
